alibi_detect/cd/utils.py

```diff
@@ -118,5 +118,7 @@ def get_input_shape(shape: Optional[Tuple], x_ref: Union[np.ndarray, list]) -> O
     else:
         logger.warning('Input shape could not be inferred. '
                        'If alibi_detect.models.tensorflow.embedding.TransformerEmbedding '
-                       'is used as preprocessing step, a saved detector cannot be reinitialized.')
+                       'is used as preprocessing step, a saved detector cannot be reinitialized. '
+                       'Additionally, if the detector is a subclassed `tf.keras.Model`, it is '
+                       'unlikely to be serializable.')
```

Comment:
Perhaps a silly question, but what is this warning about specifically? Does it just tell the user that we can't infer the input shape because the model can't be initialized? And that's because the models cannot be serialized? If that's the case should we issue a warning at the save step?